extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import 'mocha';
+import * as azdata from 'azdata';
+import * as utils from './utils';
+const path = require('path');
+import { context } from './testContext';
+import assert = require('assert');
+import { getStandaloneServer } from './testConfig';
+import { readFileSync, unlinkSync, existsSync } from 'fs';
+
+let schemaCompareService: azdata.SchemaCompareServicesProvider;
+let dacpac1: string = path.join(__dirname, 'testData/Database1.dacpac');
+let dacpac2: string = path.join(__dirname, 'testData/Database2.dacpac');
+let dummyDBName: string = 'ads_schemaCompareDB'; // This is used as fill in name and not created anywhere
+
+if (context.RunTest) {
+	suite('Schema compare integration test suite', () => {
+		suiteSetup(async function () {
+			let attemps: number = 20;
+			while (attemps > 0) {
+				schemaCompareService = await azdata.dataprotocol.getProvider<azdata.SchemaCompareServicesProvider>('MSSQL', azdata.DataProviderType.SchemaCompareServicesProvider);
+				if (schemaCompareService) {
+					break;
+				}
+				attemps--;
+				await utils.sleep(1000); // To ensure the providers are registered.
+			}
+			console.log(`Start schema compare tests`);
+		});
+		test('Schema compare dacpac to dacpac comparision and script generation', async function () {
+			assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+
+			let source: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: dacpac1,
+				databaseName: '',
+				ownerUri: '',
+			};
+			let target: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: dacpac2,
+				databaseName: '',
+				ownerUri: '',
+			};
+
+			let schemaCompareResult = await schemaCompareService.schemaCompare(source, target, azdata.TaskExecutionMode.execute);
+			assertSchemaCompareResult(schemaCompareResult);
+		});
+
+		test('Schema compare database to database comparision and script generation', async function () {
+
+			let server = await getStandaloneServer();
+			await utils.connectToServer(server, 300);
```

Comment:
I see the 300 hard code in a few places. Let make it a constant in-case it ever has to change. Plus a variable SERVER_CONNECTION_TIMEOUT is more readable.